src/background/auth_server.py

```
@@ -0,0 +1,151 @@
+        print(payload)
+        # Extract user information
+        user_email = payload.get("email")
+        user_info = json.loads(DatabaseFunctions.read_user_by_email(user_email))
+        
+        return user_info
+    except jwt.ExpiredSignatureError:
+        # Handle expired token
+        print("Token has expired")
+        return None
+    except jwt.InvalidTokenError:
+        # Handle invalid token
+        print("Invalid token")
+        return None
+def get_token(user, num_hours=1):
+    exp_time = datetime.datetime.utcnow() + datetime.timedelta(hours=num_hours)
+    return jwt.encode({
+        'email': user['email'],
+        'exp': int(exp_time.timestamp())
+    }, SECRET_KEY, algorithm="HS256")
+
+@app.route('/auth/google', methods=['POST'])
+def auth_google():
+    token = request.args.get('token', default="NO TOKEN LOADED", type=str)
+    try:
+        # Verify the token using Google's API
+        idinfo = id_token.verify_oauth2_token(token, requests.Request())
+
+        email = idinfo['email']
+
+        # Create a JWT token for the user
+        jwt_token = jwt.encode({'email': email}, API_KEY, algorithm='HS256')
+        return jsonify({'token': jwt_token})
+    except ValueError:
+        # Invalid token
+        return jsonify({'error': 'Invalid token'}), 401
+def token_required(f):
+    #Thought, what to do with current user
+    @wraps(f)
+    def decorated(*args, **kwargs):
+        token = None
+        if 'Authorization' in request.headers:
+            token = request.headers['Authorization'].split(" ")[1]  # Extract token from header
+
+        if not token:
+            return jsonify({'message': 'Token is missing!'}), 401
+
+        try:
+            user = decode_user_from_token(token)
+            current_user = DatabaseFunctions.read_user_by_email(user["email"])
+            if not current_user:
+                return jsonify({'message': 'User not found!'}), 401
```

Is this a 401, or a 500? If you successfully decoded a user from the token (this is related to my above comment you might want to throw an exception any time you cannot) but cannot find a user, something went wrong on the server side, not the client side, right?